src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,1090 @@
+  integer, private :: i_chem_som  ! index of chemically protected SOM
+  integer, private :: i_avl_som  ! index of available (aka active) SOM
+  integer, private :: i_str_lit  ! index of structural litter pool
+  integer, private :: i_l1m1  ! indices of pool transfers needed in whole module
+  integer, private :: i_l1m2
+  integer, private :: i_l2m1
+  integer, private :: i_l2m2
+  integer, private :: i_s1m1
+  integer, private :: i_s1m2
+  integer, private :: i_m1s1
+  integer, private :: i_m1s2
+  integer, private :: i_m2s1
+  integer, private :: i_m2s2
+
+  type, private :: params_type
+     ! Respiration fraction cwd --> structural litter (keep comment sep for dif)
+     real(r8):: rf_cwdl2_bgc 
+     real(r8):: tau_cwd_bgc   ! corrected fragmentation rate constant CWD, century leaves wood decomposition rates open, within range of 0 - 0.5 yr^-1 (1/0.3) (1/yr)
+     real(r8) :: minpsi_bgc   !minimum soil water potential for heterotrophic resp
+     real(r8) :: maxpsi_bgc   !maximum soil water potential for heterotrophic resp
+
+     ! TODO Need for spin-ups?
+     real(r8) :: initial_Cstocks_depth      ! Soil depth for initial Carbon stocks for a cold-start
+     real(r8), allocatable :: initial_Cstocks(:)  ! Initial Carbon stocks for a cold-start
+     real(r8), allocatable :: mge(:)  ! Microbial growth efficiency (mg/mg)
+     real(r8), allocatable :: vmod(:)  ! 
+     real(r8), allocatable :: vslope(:)  ! 
+     real(r8), allocatable :: vint(:)  ! 
+     real(r8), allocatable :: kmod(:)  ! 
+     real(r8), allocatable :: kslope(:)  ! 
+     real(r8), allocatable :: kint(:)  ! 
+     
+  end type params_type
+  !
+  type(params_type), private :: params_inst
+
+  character(len=*), parameter, private :: sourcefile = &
+       __FILE__
+
+  !-----------------------------------------------------------------------
+
+contains
+
+  !-----------------------------------------------------------------------
+  subroutine readParams ( ncid )
+    !
+    ! !DESCRIPTION:
+    !
+    ! !USES:
+    use ncdio_pio    , only: file_desc_t,ncd_io
+    !
+    ! !ARGUMENTS:
+    type(file_desc_t),intent(inout) :: ncid   ! pio netCDF file id
+    !
+    ! !LOCAL VARIABLES:
+    character(len=32)  :: subname = 'readParams'
+    character(len=100) :: errCode = 'Error reading MIMICS params '
+    logical            :: readv   ! has variable been read in or not
+    real(r8)           :: tempr   ! temporary to read in constant
+    character(len=100) :: tString ! temp. var for reading
+    !-----------------------------------------------------------------------
+
+    ! Read off of netcdf file
+    ! TODO Add new params here and in the params file.
+    ! TODO Some may need _bgc or _mimics added/removed here and in the params
+    !      file. Nicer to add as prefix instead of as suffix because params will
+    !      then appear grouped in the file. Need approval from Bill/Erik.
```

Comment:
Question for @ekluzek 
The above TODO talks about adding `bgc` or `mimics` as a **prefix** or as a **suffix** to parameter names that we want to assign to one model or the other.
@wwieder prefers the **prefix** idea because params will then appear grouped in the file. Before I go ahead and implement this, do you object by any chance?

Reply:
Mostly I suggest to go with Will's suggestion as he's going to be working with these variables, and that's a good note about the prefix as far as grouping the variables. There are two caveat's I have to do that though. The first is wondering about our current use of prefixes vs suffixes and if we have any conventions there. The second is that if you do this for MIMICS_ you should likely do the same for SOILBGC_ for the same reason.

In terms of prefixes and suffixes in the history variables. We do have a convention for a prefix that gives the species type (C13_ or C14_ for carbon isotopes, and also CH4_ and O_ for Methane, O3 for Ozone, H2O in some cases for water). In terms of model packages we do have _FUN suffixes for a few variables. A leading T is usually used in TemperatureType for temperatures. SNO for snow variables, SOIL for something to do with soil. Then there's pool types done as prefixes: ACT_, CEL_, DWT_, F for radiation fluxes, Q for water fluxes of various types, CONC_ for some concentrations. There's not a strong enough convention to blindly follow it, so doing something different could be OK here.

Reply:
Thanks @wwieder and @ekluzek . I will work on that.

Here's how I understand what I'm doing. **Pls correct anything that I have misunderstood:**
- In the params file(s) I will add prefixes `mimics_*` or `bgc_*` to model-specific parameters.
- In the code I will read model-specific parameters in the corresponding module `*MIMICSMod` or `*BGCMod`.
- In the params file(s) I will remove such prefixes (or suffixes as the case may be) from parameters that are used by both DecompCascade models.
- In the code I will read parameters that are used by both models in `CNSharedParamsMod`.
- **This one is a clarification question:** In the code, once parameters have been read from the params file(s), do we still want the parameter names to include the prefixes to maintain clarity about where they came from, or is it sufficient that they have prefixes in the params file(s)? I ask because I found myself leaning in favor of removing the prefixes in the parameter names once we're using them in the code. But maybe that's wrong...

Reply:
Interesting question on whether to remove the prefix in the code. I don't have strong feelings on this, but I have heard from a few people that they prefer if names don't change from one place to another, to make it easier to grep for things in the code. It seems like this is a situation where that would apply: It's a bit easier to find uses of a parameter if the same name is used in the code as in the param file (otherwise you need to first figure out that it has changed name, then grep for the name that is actually used in the code).

That said, I also see the rationale for removing the prefixes to keep the code more concise and easier to read.

Here's a possible compromise solution that comes to mind: We use a separator other than `_` to separate the prefix from the main variable name in the params file. We choose a separator that is reserved for this purpose of "namespacing" the variables in the params file, and isn't used in any other way in parameter names (maybe `-` or `.`, if they are allowed in NetCDF variable names, or something else). Then we can strip that prefix in the variable name in the code. This would let us have a consistent rule like, "everything before the dot in the param file name is just for namespacing purposes, so to find the variable in the code, just grep for the name after the dot".

Reply:
Thank you for this suggestion, @billsacks .  I opted for the more immediately straightfwd solution of keeping the prefixes.